src/probes/klog_main.c

```diff
@@ -59,7 +59,7 @@ int main(void)
                 bufp = (char *)calloc(buflen, sizeof(char));
 
                 malloc_trim(0);
-                bytes = klogctl(SYSLOG_ACTION_READ, bufp, log_size);
+                bytes = klogctl(SYSLOG_ACTION_READ, bufp, (int)buflen);
```

Comment:
Good catch. MAX_BUF is supposed to be a limit on the number of bytes we read, so this is an escape. I think the intention was to ++loopcount. I'll separate that out of the conditional statement for clarity.